src/renderer/widgets/CreateWallet/ui/MultisigWallet/components/Signatory.tsx

```diff
@@ -170,74 +142,91 @@ export const Signatory = ({
     });
   };
 
-  useEffect(() => {
-    if (displayName && displayName !== signatoryName) {
-      onNameChange(displayName);
-    }
-  }, [displayName]);
-
-  const onAddressChange = (data: ComboboxOption) => {
+  const onAddressChange = (address: AccountAddress) => {
     if (!chain) return;
 
     const isEthereumChain = networkUtils.isEthereumBased(chain.options);
     const validateFn = isEthereumChain ? validateEthereumAddress : validateSubstrateAddress;
 
-    const validatedAddress = validateFn(data.value) ? data.value : '';
-    const fixedAddress = toAddress(validatedAddress, { prefix: chain?.addressPrefix });
+    if (!validateFn(address)) return;
+
+    const wallet = walletUtils.getWalletFilteredAccounts(wallets, {
+      accountFn: ({ accountId }) => accountId === toAccountId(address),
+    });
```

Comment:
we need to check for validSignatory here as well or we could have a proxy wallet